sonar-scanner-engine/src/main/java/org/sonar/scanner/issue/IssueTransformer.java

```diff
@@ -52,7 +52,8 @@ public static TrackedIssue toTrackedIssue(ServerIssue serverIssue) {
     issue.setEndLine(serverIssue.hasLine() ? serverIssue.getLine() : null);
     issue.setSeverity(serverIssue.getSeverity().name());
     issue.setAssignee(serverIssue.hasAssigneeLogin() ? serverIssue.getAssigneeLogin() : null);
-    issue.setComponentKey(ComponentKeys.createEffectiveKey(serverIssue.getModuleKey(), serverIssue.hasPath() ? serverIssue.getPath() : null));
+    String keyWithoutBranch = StringUtils.substringBefore(serverIssue.getModuleKey(), ":");
```

Comment:
AFAIR, project key may contain ":" while branch cannot. For example default key for Maven project is <groupId>:<artifactId>. So you should use StringUtils.substringBeforeLast